### PR TITLE
Inflatable station

### PR DESCRIPTION
**Pull request by MadmanMartian**
REQUIRES #21393 TO BE MERGED FIRST

An idea tossed around in #ick a while ago

:cl:
 * rscadd: Adds Discount Dans inflatable station in a can.

### CODE REVIEW

**Comment by Durgen** [comment]
webm?

**Comment by Goldsteel** [comment]
I imagine this would be really cool for events where big bombs go off or the singulo gets loose, but as it was mentioned before, it should be noted that it does introduce the idea of instant space cult bases with no effort required.

If we had base-in-a-can then it should maybe be restricted to ERT or other responders that might need to make a base in a hellscape.

Oh, and, since it's DanCo we should probably add some unfortunate side effects like it being filled with spiderlings, burned mess, cheese, dan pies, something like that that got 'misplaced' with low probability.

**Comment by MadmanMartian** [comment]
way ahead of you.

May make this a bit more OOP, make instant room-gen function that can be fed types.

File: code/game/objects/items/weapons/grenades/inflatable.dm [hunk 1]
```diff
@@ -13,4 +13,32 @@
...
+/**
+	Makes an instant inflatable station
+**/
+/obj/item/weapon/grenade/inflatable/station/prime()
+	playsound(src, 'sound/items/zip.ogg', 75, 1)
+	var/list/possible_trash = subtypesof(/obj/item/trash)-typesof(/obj/item/trash/mannequin)
+	var/turf/source = get_turf(src)
+	for(var/mob/living/M in view(source, range))
+		if(get_turf(M) == source)
+			M.gib()
```
Comment (Kurfursten) [inline code/game/objects/items/weapons/grenades/inflatable.dm:33]: This is pretty scary if you ever intend for this to be something non-bus

File: code/game/objects/items/weapons/grenades/inflatable.dm [hunk 2]
```diff
@@ -13,4 +13,32 @@
...
+			continue
+		to_chat(M, "<span class = 'warning'>You are bounced away from \the [src] as it deploys!</span>")
+		M.throw_at(get_ranged_target_turf(source, get_dir(source, M), range*3), 50, 3)
```
Comment (Kurfursten) [inline code/game/objects/items/weapons/grenades/inflatable.dm:36]: Mind estimating how damaging this is just being thrown against a wall?

File: code/modules/procedural mapping/instant_room.dm [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	var/list/affected_turfs
+	var/list/internal_turfs
+	switch(shape)
+		if("circle")
+			affected_turfs = circleviewturfs(source, range)
+			internal_turfs = circleviewturfs(source, range-1)
+		else
+			affected_turfs = range(source, range)
+			internal_turfs = range(source, range-1)
+	for(var/turf/T in affected_turfs)
```
Comment (Kurfursten) [inline code/modules/procedural mapping/instant_room.dm:11]: couldn't we speed this up by removing internal_turfs from affected_turfs, then just looping through each once?

it avoids running a comparison of list contents a lot of times

**Pull request closed without merging**